server/db/driver/pg/types.go

```diff
@@ -0,0 +1,45 @@
+package pg
+
+import (
+	"fmt"
+)
+
+// fastUint64 provides an efficient Scanner for 64-bit integer data. With a
+// regular uint64 or any other integer type that itself does not implement
+// Scanner, the database/sql.Scan implementation will coerce type by converting
+// to and from a string, so we avoid this expensive operation by implementing a
+// Scanner that uses a type assertion. This only works for columns with a
+// postgresql type of INT8 (8-byte integer) that scan a value of type int64.
+type fastUint64 uint64
+
+func (n *fastUint64) Scan(value interface{}) error {
+	if value == nil {
+		*n = 0
+		return fmt.Errorf("NULL not supported")
+	}
+
+	v, ok := value.(int64)
+	if !ok {
+		return fmt.Errorf("not a signed 64-bit integer: %T", value)
+	}
+	*n = fastUint64(v)
+	return nil
+
+	// NOTE: If we want to be able to scan all other interger types and coerce
+	// them into uint64, we can switch on each signed type. PostgreSQL has no
+	// unsigned types, so just int8, int16, int32, and int64.
```

Comment:
Yeah it's a bit silly.  In our code we use uint64 because most cases negative is not sensible, but postgresql just doesn't support it.  Since this type just scans, reading from pg to Go it's fine as long as the PG table didn't contain a negative number.  If that's the case, same issue with scanning into a regular uint64 instead of fastUint64.

Reply:
I see. But I meant, can we write/store a uint64? In reality, we are limited to saving an Int64.

Reply:
Correct, even with `uint64` instead of `fastuint64`, we can't actually store it the highest values.  Well, I suspect pg would make it negative... anyway, this type is just for scanning as it doesn't have a Value method.